chainstate/src/detail/block_index_history_iter.rs

```diff
@@ -63,51 +63,128 @@ impl<'a, H: BlockIndexHandle> Iterator for BlockIndexHistoryIterator<'a, H> {
 
 #[cfg(test)]
 mod tests {
+    use super::*;
+    use chainstate_storage::inmemory::Store;
+    use common::chain::block::timestamp::BlockTimestamp;
+    use common::chain::block::BlockReward;
+    use common::chain::block::ConsensusData;
+    use common::chain::config::Builder as ChainConfigBuilder;
+    use common::chain::config::ChainType;
+    use common::chain::Block;
+    use common::chain::Destination;
+    use common::chain::NetUpgrades;
+    use common::primitives::id::WithId;
+    use common::primitives::time;
     use common::primitives::{Idable, H256};
+    use common::time_getter::TimeGetter;
+    use std::sync::Arc;
 
-    use crate::{detail::tests::TestFramework, BlockSource};
-
-    use super::*;
+    use crate::detail::Chainstate;
+    use crate::BlockSource;
+    use crate::ChainstateConfig;
 
     #[test]
     fn history_iteration() {
         utils::concurrency::model(|| {
-            let mut tf = TestFramework::default();
+            let chain_config = ChainConfigBuilder::new(ChainType::Mainnet)
```

Comment:
Because we can't use the test framework as a dependency in chainstate because it'll cause circular dependency. We either move this to the test framework crate or we make it independent of it. One of these have to happen.